music_assistant/providers/bbc_sounds/__init__.py

```diff
@@ -0,0 +1,405 @@
+            label="Password",
+            required=False,
+        ),
+        ConfigEntry(
+            key="show_local",
+            category="advanced",
+            type=ConfigEntryType.BOOLEAN,
+            label="Show local radio stations?",
+            default_value=False,
+        ),
+        ConfigEntry(
+            key="now_playing",
+            category="advanced",
+            type=ConfigEntryType.BOOLEAN,
+            label="Show 'now playing' details?",
+            description=(
+                "Show details of the currently playing track instead of the stationdetails"
+            ),
+            default_value=True,
+        ),
+        ConfigEntry(
+            key="update_interval",
+            category="advanced",
+            type=ConfigEntryType.INTEGER,
+            label="Player update interval (seconds)",
+            description="How often to check for now playing updates",
+            default_value=5,
+            range=(1, 60),
+            depends_on="now_playing",
+            depends_on_value=True,
+        ),
+    )
+
+
+class BBCSoundsProvider(MusicProvider):
+    """A MusicProvider class to interact with the BBC Sounds API via auntie-sounds."""
+
+    # This is the image id that is shown when there's no track image
+    BLANK_IMAGE_NAME = "p0bqcdzf"
+    client: SoundsClient
+
+    async def handle_async_init(self) -> None:
+        """Handle async initialization of the provider."""
+        self.client = SoundsClient(
+            session=self.mass.http_session,
+            logger=self.logger,
+        )
+
+        # If we have an account, authenticate. Testing shows all features work without auth
+        # but BBC will be disabling BBC Sounds from outside the UK at some point
+        if self.config.get_value(CONF_USERNAME) and self.config.get_value(CONF_PASSWORD):
+            try:
+                await self.client.auth.authenticate(
+                    username=str(self.config.get_value(CONF_USERNAME)),
+                    password=str(self.config.get_value(CONF_PASSWORD)),
+                )
+            except exceptions.LoginFailedError as e:
+                raise LoginFailed(e)
+
+    async def loaded_in_mass(self) -> None:
+        """Call after the provider has been loaded."""
+        self.mass.create_task(self._schedule_now_playing_updates())
+        return await super().loaded_in_mass()
+
+    async def _update_player_now_playing(self, queue_id: str, station_id: str) -> None:
+        """Manually update a player's now playing information."""
+        # TODO: detect blank song images
+        queue = self.mass.player_queues.get(queue_id=queue_id)
+        if not queue:
+            return
+        station = await self.client.stations.get_station(station_id=station_id)
+        if not station:
+            return
+        now_playing = await self.client.schedules.currently_playing_song(station, image_size=1280)
+
+        # TODO: check if there is a neater way to do this
```

Comment:
The better way to do this is from "get_streamdetails". So you hand out the streamdetails and then you keep updating the stream metadata. You may not interact with the queue or queue item directly.

Just launch a task from the "get_streamdetails" that keeps updating the stream metadata.
And then the best way to break out of it is when the streamdetails have a "seconds_streamed" filled filled because that means the stream was done.